src/DataStructures/Python/ToNumpyArray.cpp

```diff
@@ -0,0 +1,31 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#include "DataStructures/Python/ToNumpyArray.hpp"
+
+#include <array>
+#include <cstddef>
+#include <utility>
+#include "DataStructures/Python/Numpy.hpp"
+
+#include "DataStructures/Matrix.hpp"
+
+namespace py_bindings {
+
+PyObject* to_numpy_array(const Matrix& matrix) {
+  double* c_style_data = static_cast<double*>(
+      malloc(sizeof(double) * (matrix.rows()) * matrix.columns()));
+  for (size_t i = 0; i < matrix.rows(); ++i) {
+    for (size_t j = 0; j < matrix.columns(); ++j) {
+      c_style_data[j + i * matrix.columns()] = matrix(i, j);
+    }
+  }
+  std::array<long, 2> dims{
+      {static_cast<long>(matrix.rows()), static_cast<long>(matrix.columns())}};
+  PyObject* npy_array =
+      PyArray_SimpleNewFromData(2, dims.data(), NPY_DOUBLE, c_style_data);
+  PyArrayObject* npy_array_obj = reinterpret_cast<PyArrayObject*>(npy_array);
+  PyArray_ENABLEFLAGS(npy_array_obj, NPY_ARRAY_OWNDATA);
+  PyObject* npy_new_array = reinterpret_cast<PyObject*>(npy_array_obj);
```

Comment:
I don't think you need this last cast, I think you can do:
```cpp
  PyArray_ENABLEFLAGS(npy_array_obj, NPY_ARRAY_OWNDATA);
  return npy_arry;
```
The reason is that after the first `reinterpret_cast` the pointer still points to the original object, and so you are mutating the original object that is also pointed to by `npy_array` (this is why we generally avoid pointers whenever possible :) )